Content.Shared/Throwing/ThrownItemSystem.cs

```diff
@@ -114,18 +119,25 @@ private void StopThrow(EntityUid uid, ThrownItemComponent thrownItemComponent)
                 }
             }
 
-            EntityManager.EventBus.RaiseLocalEvent(uid, new StopThrowEvent {User = thrownItemComponent.Thrower}, true);
+            var stopThrow = new StopThrowEvent(thrownItemComponent.Thrower);
+            EntityManager.EventBus.RaiseLocalEvent(uid, ref stopThrow, true);
             EntityManager.RemoveComponent<ThrownItemComponent>(uid);
         }
 
-        public void LandComponent(ThrownItemComponent thrownItem)
+        /// <summary>
+        ///     Called when thrownItem should hit the ground and slide, can stop movement entirely if specified.
+        /// </summary>
+        public void LandComponent(ThrownItemComponent thrownItem, bool stopMoving = false)
         {
             if (thrownItem.Deleted || Deleted(thrownItem.Owner) || _containerSystem.IsEntityInContainer(thrownItem.Owner)) return;
 
             var landing = thrownItem.Owner;
 
+            if (stopMoving && _entityManager.TryGetComponent(landing, out IPhysBody? physics))
```

Comment:
proxy method

Reply:
I _think_ I used proxy methods everywhere I could with the new commits, can't seem to just do TryGetComponent on its own, got rid of resolving EntityManager but it still needs EntityManager.TryGetComponent in this particular instance

Reply:
You can use TryComp<>() proxy method